crates/pathfinder/src/config.rs

```diff
@@ -1001,29 +1001,41 @@ impl ConsensusConfig {
 #[cfg(feature = "p2p")]
 impl ConsensusConfig {
     fn parse_or_exit(args: ConsensusCli) -> Option<Self> {
-        if args.validator_addresses.len() < 2 {
-            Cli::command()
-                .error(
-                    clap::error::ErrorKind::ValueValidation,
-                    "At least 3 validator addresses are required (including this node's address).",
-                )
-                .exit();
-        }
-
-        args.is_enabled.then_some(Self {
-            proposer_address: ContractAddress(
-                args.proposer_address
-                    .expect("Clap requires this to be set if `is_enabled` is true"),
-            ),
-            my_validator_address: ContractAddress(
-                args.my_validator_address
+        args.is_enabled.then(|| {
+            if std::iter::once(
+                &args
+                    .my_validator_address
                     .expect("Clap requires this to be set if `is_enabled` is true"),
-            ),
-            validator_addresses: args
-                .validator_addresses
-                .into_iter()
-                .map(ContractAddress)
-                .collect(),
+            )
+            .chain(args.validator_addresses.iter())
+            .collect::<HashSet<_>>()
+            .len()
+                < 3
+            {
+                Cli::command()
+                    .error(
+                        clap::error::ErrorKind::ValueValidation,
+                        "At least 3 unique validator addresses are required (including this \
+                         node's validator address).",
```

Comment:
Why would we want to include this validator's address?

Reply:
Ok I think I get it now. You mean at least 3 required, and self counts. I thought you had to include the self address here, which made no sense.